### PR TITLE
Adding distributed layernorm to llama prefill

### PR DESCRIPTION
**Pull request by avoraTT**
### Ticket
- #8796 

### What's changed
- Replacing sharded rmsnorm with distributed rmsnorm in prefill before attention, mlp, and lm head 
- created sharded counterparts to norm, attn_norm, and ffn_norm

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)

### CODE REVIEW

**Comment by avoraTT** [comment]
- [ ] Frequent: https://github.com/tenstorrent/tt-metal/actions/runs/9861834533
- [ ] Demo: https://github.com/tenstorrent/tt-metal/actions/runs/9861909959
- [ ] Perf: https://github.com/tenstorrent/tt-metal/actions/runs/9861840979
- [ ] Post: https://github.com/tenstorrent/tt-metal/actions/runs/9861916696

File: models/demos/t3000/llama2_70b/tt/llama_decoder_optimized.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def __init__(
         self.device_mesh = device_mesh
         self.num_devices = device_mesh.get_num_devices()
         self.model_config = model_config
+        read_cache = False
```
Comment (cglagovichTT) [inline models/demos/t3000/llama2_70b/tt/llama_decoder_optimized.py:35]: I think this should not be merged

File: models/demos/t3000/llama2_70b/tt/llama_decoder_optimized.py [hunk 2]
```diff
@@ -105,6 +109,17 @@ def load_weights(self):
         )
         self.attn_norm = ttnn.to_device(attn_norm_ttnn, self.device_mesh)
```
Comment (cglagovichTT) [inline models/demos/t3000/llama2_70b/tt/llama_decoder_optimized.py:110]: Can we delete the old norm tensors now?

File: models/demos/t3000/llama2_70b/tt/llama_decoder_optimized.py [hunk 3]
```diff
@@ -116,6 +131,17 @@ def load_weights(self):
         )
         self.ffn_norm = ttnn.to_device(ffn_norm_ttnn, self.device_mesh)
```
Comment (cglagovichTT) [inline models/demos/t3000/llama2_70b/tt/llama_decoder_optimized.py:132]: Same here, delete this?

File: models/demos/t3000/llama2_70b/tt/llama_model_optimized.py [hunk 4]
```diff
@@ -39,6 +39,7 @@ def __init__(
         self.device_mesh = device_mesh
         self.num_devices = device_mesh.get_num_devices()
         self.model_config = model_config
+        read_cache = False
```
Comment (cglagovichTT) [inline models/demos/t3000/llama2_70b/tt/llama_model_optimized.py:42]: can remove this?

File: models/demos/t3000/llama2_70b/tt/llama_model_optimized.py [hunk 5]
```diff
@@ -138,6 +140,17 @@ def load_weights(self):
         )
         self.norm = ttnn.to_device(norm_ttnn, self.device_mesh)
```
Comment (cglagovichTT) [inline models/demos/t3000/llama2_70b/tt/llama_model_optimized.py:141]: Same here, can this be deleted?

**Review by cglagovichTT — APPROVED** [review]
Looks good. Approved so you can merge after addressing comments

**Pull request merged**